src/core/src/main/java/org/locationtech/geogig/model/RevFeatureBuilder.java

```diff
@@ -71,10 +76,33 @@ public RevFeatureBuilder addProperty(Property featureProp) {
      */
     public RevFeatureBuilder addValue(@Nullable Object value) {
         value = safeCopy(value);
+        if (value instanceof Geometry) {
+            value = normalizeIfNeeded((Geometry) value);
+        }
         this.values.add(value);
         return this;
     }
 
+    Geometry normalizeIfNeeded(Geometry value) {
+        if (value instanceof Polygon) {
+            value.normalize();
+        } else if (value instanceof MultiPolygon
```

Comment:
`GeometryCollection.class.isAssignableFrom` would catch MultiPoint and MultiLineString too, which I don't want, hence the check for Multipolygon or the more generic GeometryCollection (like in a GeometryCollection that contains a mix of other geometry types).